diff --git a/src/sofa/collisionAlgorithm/BaseAABBBroadPhase.h b/src/sofa/collisionAlgorithm/BaseAABBBroadPhase.h
index faa84fa..7f59d00 100644
--- a/src/sofa/collisionAlgorithm/BaseAABBBroadPhase.h
+++ b/src/sofa/collisionAlgorithm/BaseAABBBroadPhase.h
@@ -26,8 +26,8 @@ public:
     , d_static(initData(&d_static, false,"isStatic", "Optimization: object is not moving in the scene"))
     , d_method(initData(&d_method, 0,"method", "chosen method to determine the boxes containing the elements"))
     , d_thread(initData(&d_thread, 8,"thread","Number of threads")){
-        c_nbox.addInputs({&d_nbox});
-        c_nbox.addCallback(std::bind(&BaseAABBBroadPhase::updateBroadPhase,this));
+        // c_nbox.addInputs({&d_nbox});
+        // c_nbox.addCallback(std::bind(&BaseAABBBroadPhase::updateBroadPhase,this));
     }
 
     virtual void newContainer() = 0;
@@ -61,6 +61,7 @@ public:
     }
 
     inline void doUpdate() {
+        if (!l_geometry) return;
         sofa::helper::AdvancedTimer::stepBegin("========================= AABBBroadPhase do update =========================");
         m_Bmin = l_geometry->getPosition(0);
         m_Bmax = m_Bmin;
diff --git a/src/sofa/collisionAlgorithm/algorithm/FindClosestProximityAlgorithm.h b/src/sofa/collisionAlgorithm/algorithm/FindClosestProximityAlgorithm.h
index 4c73e70..2a3888f 100644
--- a/src/sofa/collisionAlgorithm/algorithm/FindClosestProximityAlgorithm.h
+++ b/src/sofa/collisionAlgorithm/algorithm/FindClosestProximityAlgorithm.h
@@ -21,8 +21,8 @@ public:
 //    Data<sofa::type::vector<double> > d_outputDist;
 
     FindClosestProximityAlgorithm()
-    : l_from(initLink("from", "link to from geometry"))
-    , l_dest(initLink("dest", "link to dest geometry"))
+    : l_from(initLink("fromGeom", "link to from geometry"))
+    , l_dest(initLink("destGeom", "link to dest geometry"))
     , d_drawCollision (initData(&d_drawCollision, true, "drawcollision", "draw collision"))
     , d_output(initData(&d_output,"output", "output of the collision detection"))
     , d_projective(initData(&d_projective, false,"projective", "projection of closest prox onto from element"))
